File: lib/domain/repositories/asistencia_repository.dart
import '../entities/registro_asistencia.dart';
import '../entities/curso.dart';

abstract class AsistenciaRepository {
  Future<List<RegistroAsistencia>> getRegistrosPorCursoYFecha(
    String cursoId,
    DateTime fecha,
  );

  Future<ResumenAsistencia> getResumenDiario(DateTime fecha);

  Future<ResumenAsistencia> getResumenPorTurno(
    String turno,
    DateTime fecha,
  );

  Future<RegistroAsistencia> registrarIngresoManual({
    required String alumnoId,
    required String cursoId,
    required DateTime horaIngreso,
  });

  Future<RegistroAsistencia> registrarRetiro({
    required String registroId,
    required DateTime horaRetiro,
    required String motivo,
  });

  Future<RegistroAsistencia> marcarNoComputable({
    required String registroId,
    required String motivo,
  });
}

File: backend/app/domain/repositories/asistencia_repository.py
from abc import ABC, abstractmethod
from typing import List
from datetime import date
from app.domain.entities.registro_asistencia import RegistroAsistencia

class AsistenciaRepository(ABC):

    @abstractmethod
    def get_asistencia_diaria(self, curso_id: str, fecha: date) -> List[RegistroAsistencia]: ...

    @abstractmethod
    def crear_registro(self, registro: RegistroAsistencia) -> RegistroAsistencia: ...